clconf/template_test.go

```diff
@@ -0,0 +1,453 @@
+		{fullPath: filepath.Join(temp, "yes_basedir2.html"+extension), relPath: "yes_basedir2.html" + extension},
+	}
+
+	if !reflect.DeepEqual(paths, expected) {
+		t.Errorf("Path didn't match expected [%v] != [%v]", paths, expected)
+	}
+}
+
+func TestFindTemplatesWithoutExtension(t *testing.T) {
+	extension := ""
+	temp := buildTestFolder(t, extension)
+	defer os.RemoveAll(temp)
+
+	paths, err := findTemplates(temp, extension)
+	if err != nil {
+		t.Fatalf("Error running findTemplates: %v", err)
+	}
+
+	normalizePaths(paths)
+
+	expected := []pathWithRelative{
+		{fullPath: filepath.Join(temp, "no_basedir.html"), relPath: "no_basedir.html"},
+		{fullPath: filepath.Join(temp, "no_basedir1.html"), relPath: "no_basedir1.html"},
+		{fullPath: filepath.Join(temp, "subdir1/no_subdir1.sh"), relPath: "subdir1/no_subdir1.sh"},
+		{fullPath: filepath.Join(temp, "subdir1/subsubdir1/no_subdir1subsubdir1.sh"), relPath: "subdir1/subsubdir1/no_subdir1subsubdir1.sh"},
+		{fullPath: filepath.Join(temp, "subdir1/subsubdir1/yes_subdir1subsubdir1.sh"+extension), relPath: "subdir1/subsubdir1/yes_subdir1subsubdir1.sh" + extension},
+		{fullPath: filepath.Join(temp, "subdir1/subsubdir2/no_subdir1subsubdir2.sh"), relPath: "subdir1/subsubdir2/no_subdir1subsubdir2.sh"},
+		{fullPath: filepath.Join(temp, "subdir1/subsubdir2/yes_subdir1subsubdir2.sh"+extension), relPath: "subdir1/subsubdir2/yes_subdir1subsubdir2.sh" + extension},
+		{fullPath: filepath.Join(temp, "subdir1/yes_subdir1.sh"+extension), relPath: "subdir1/yes_subdir1.sh" + extension},
+		{fullPath: filepath.Join(temp, "subdir2/no_subdir2.sh"), relPath: "subdir2/no_subdir2.sh"},
+		{fullPath: filepath.Join(temp, "subdir2/yes_subdir2.sh"+extension), relPath: "subdir2/yes_subdir2.sh" + extension},
+		{fullPath: filepath.Join(temp, "yes_basedir.html"+extension), relPath: "yes_basedir.html" + extension},
+		{fullPath: filepath.Join(temp, "yes_basedir1.html"+extension), relPath: "yes_basedir1.html" + extension},
+		{fullPath: filepath.Join(temp, "yes_basedir2.html"+extension), relPath: "yes_basedir2.html" + extension},
+	}
+
+	if !reflect.DeepEqual(paths, expected) {
+		t.Errorf("Path didn't match expected [%v] != [%v]", paths, expected)
+	}
+}
+
+func TestFindTemplatesSingleFileMatchingExtension(t *testing.T) {
+	extension := ".clconf"
+	temp := buildTestFolder(t, extension)
+	defer os.RemoveAll(temp)
+
+	paths, err := findTemplates(filepath.Join(temp, "subdir1/subsubdir1/yes_subdir1subsubdir1.sh"+extension), extension)
+	if err != nil {
+		t.Fatalf("Error running findTemplates: %v", err)
+	}
+
+	normalizePaths(paths)
+
+	expected := []pathWithRelative{
+		{fullPath: filepath.Join(temp, "subdir1/subsubdir1/yes_subdir1subsubdir1.sh"+extension), relPath: "yes_subdir1subsubdir1.sh" + extension},
+	}
+
+	if !reflect.DeepEqual(paths, expected) {
+		t.Errorf("Path didn't match expected [%v] != [%v]", paths, expected)
+	}
+}
```

Comment:
this whole test idiom is super repetitive.  i dont think any clarity would be lost if you:
```go
type relWithExt struct {
    rel string,
    ext string
}

func assertFindTemplates(t *testing.T, message, path, extension string, expected []relWithExt) {
	temp := buildTestFolder(t, extension)
	defer os.RemoveAll(temp)

	paths, err := findTemplates(filepath.Join(temp, path+extension), extension)
	if err != nil {
		t.Fatalf("%s: Error running findTemplates: %v", message, err)
	}

	normalizePaths(paths)

	normalizedExpected = []pathWithRelative{}
	for _, item := range relWithExt {
		relPath=item["rel"]+item["ext"]
		normalizedExpected = append(
			normalizedExpected,
			pathWithRelative{
				fullPath: filepath.Join(temp, relPath),
				relPath: relPath,
			})
	}

	if !reflect.DeepEqual(paths, expected) {
		t.Errorf("%s: Path didn't match expected [%v] != [%v]", message, paths, expected)
	}
}
```
then your tests become:
```go
func TestFindTemplatesSingleFileMatchingExtension(t *testing.T) {
	assertFindTemplates(t,
		"find templates single file matching extension",
		[]relWithExt{"subdir1/subsubdir1/yes_subdir1subsubdir1.sh", extension}
	}
}
```
it does _hide_ the call to `findTemplates` but the `assertFindTemplates` makes that obvious (and is similar to other [clconf tests](https://github.com/pastdev/clconf/blob/master/clconf/clconf_test.go#L98-L123))

Reply:
[Fixed](https://github.com/pastdev/clconf/pull/11/commits/280893401f46c28c25ee0cb1b217ffb0469a2ff1)